lib/modules/dosomething/dosomething_user/dosomething_user.module

```diff
@@ -103,9 +103,36 @@ function dosomething_user_menu() {
     'file' => 'dosomething_user.admin.inc',
     'weight' => 500,
   );
+
+    return $items;
+}
+
+function dosomething_user_menu_alter(&$items) {
+  unset($items['user']);
+  unset($items['user/%user']);
+  $items['user/%'] = array(
+    'page callback' => 'dosomething_user_get_view',
+    'page arguments' => array(1),
+    'access callback' => TRUE,
+  );
+  unset($items['user/%user']);
+
   return $items;
 }
 
+function dosomething_user_get_view($id) {
+  if (is_numeric($id)) {
+    $account = user_load($id);
+    return user_view($account);
+  }
+  else {
+    $phoenix_user = dosomething_user_get_user_by_northstar_id($id);
+    if (!$phoenix_user) {
+      drupal_not_found();
+    }
+    drupal_goto('user/' . $phoenix_user->uid);
```

Comment:
@aaronschachter The profile template is only wrapped in the default account page header with `return user_view($phoenix_user)`, if I use `drupal_goto` it looks correct, sorry for the confusion. 

I think ideally we'd want to use `user_view` though, in which case good call on where to put the return.